platform/core.windows/src/org/netbeans/core/windows/services/NbPresenter.java

```diff
@@ -1599,7 +1584,10 @@ private Window findFocusedWindow() {
                 }
             }
         }
-        while( null != w && !w.isShowing() ) {
+        while( null != w ) {
+            if ((w instanceof Frame || w instanceof Dialog) && w.isShowing()) {
+                break;
+            }
```

Comment:
This was a deliberate, if ancillary, addition.  While testing the problem, I noticed the bounds of the dialog are wrong (and on another screen for me) because it's using the popup window that was being found as parent. Dialog from status bar button equivalent is in the right place.  So added the same parenting logic here to work out what to calculate bounds from.

Reply:
I wonder if what you're seeing is the same as
https://github.com/apache/netbeans/pull/5280#issuecomment-1384547838

I've never been able to reproduce it.

Reply:
Not sure it is.  From the screenshot there, that is not triggered from a popup window.  I didn't look too much in depth at why the calculations are off in that cases - maybe because it's a popup, or because it's smaller than the dialog??  Anyway, using the same parenting logic to find the closest owning frame or dialog to calculate the bounds makes sense to me.

Reply:
Sorry, somehow missed that, and concentrated on the "not a popup" comment in https://github.com/apache/netbeans/pull/5280#pullrequestreview-1252234566 Could very well be the same issue then!

Reply:
Just checked, and yes I can reproduce the issue with dialog placement and the editor tab dropdown, and can confirm this change fixes that too.

Reply:
For NB20+ this could probably just use the parent passed in to the constructor now with these changes?